provider/internal/device/asa/resource.go

```diff
@@ -201,7 +206,10 @@ func (r *AsaDeviceResource) Read(ctx context.Context, req resource.ReadRequest,
 	stateData.SocketAddress = types.StringValue(asaReadOutp.SocketAddress)
 	stateData.Host = types.StringValue(asaReadOutp.Host)
 	stateData.IgnoreCertificate = types.BoolValue(asaReadOutp.IgnoreCertificate)
-	stateData.Labels = util.GoStringSliceToTFStringList(asaReadOutp.Tags.Labels)
+	// only set labels if it is different
```

Comment:
wondering if we want to add a test to make sure we are not set labels if they are the same

Reply:
done 👍